### ISSUE TITLE
Documentation links are hard to spot

### ISSUE THREAD

**Issue by thibaudcolas** [Microsoft/TypeScript#28126]
On the TypeScript website (e.g. https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html), links within the page content are hard to spot. There are two issues:

- Normal links are different from paragraph text in color only, with a contrast ratio to surrounding text that is too low (text: `#242424`, link: `#265988`, contrast ratio: [`2.11:1`](https://contrast-ratio.com/#%23265988-on-%23242424)).
- Links on `code`-style text are visually identical to `code`-style text without links

Example:

<img width="762" alt="typescript-website-link-styles" src="https://user-images.githubusercontent.com/877585/47487755-af4ce880-d84b-11e8-8565-09e925e81ff9.png">

**Search Terms:**

Website links, documentation links, documentation code link, accessibility

**Expected behavior:**

It should be possible to visually scan a documentation page and identify what is a link.

**Actual behavior:**

Links on paragraph text have a very low contrast ratio that makes them hard to spot. Links on code are impossible to spot, and it's very frustrating to have to hover all the inline snippets or try to guess which ones are links.

**Related Issues:** Couldn't find any.

---

I can see a few solutions to this:

- Style links based on something else than color. The established accessibility best practice is to use an underline.
- Pick a color for links that has high-enough contrast to surrounding text.
- For code links, display them with the link color rather than the code color `#bf414a`.

Relevant resources:

- https://webaim.org/techniques/hypertext/link_text#appearance
- [WCAG 2.0 F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/TR/WCAG20-TECHS/F73.html)
- [WCAG 2.0 G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/TR/WCAG20-TECHS/G183.html)
